src/components/GasFeeWarning.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet } from 'react-native'
+import AppAnalytics from 'src/analytics/AppAnalytics'
+import { AppEvents } from 'src/analytics/Events'
+import InLineNotification, { NotificationVariant } from 'src/components/InLineNotification'
+import { CICOFlow } from 'src/fiatExchanges/types'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { Spacing } from 'src/styles/styles'
+import { PreparedTransactionsResult } from 'src/viem/prepareTransactions'
+
+export type GasFeeWarningFlow = 'Send' | 'Swap' | 'Withdraw' | 'Deposit' | 'Dapp'
+
+function GasFeeWarning({
+  prepareTransactionsResult,
+  flow,
+  onPressSmallerAmount,
+}: {
+  prepareTransactionsResult?: PreparedTransactionsResult
+  flow: GasFeeWarningFlow
+  onPressSmallerAmount?: (amount: string) => void
+}) {
+  const { t } = useTranslation()
+
+  const feeCurrency = prepareTransactionsResult
+    ? prepareTransactionsResult.type === 'not-enough-balance-for-gas'
+      ? prepareTransactionsResult.feeCurrencies[0]
+      : prepareTransactionsResult.feeCurrency
+    : undefined
+
+  useEffect(() => {
+    if (feeCurrency && prepareTransactionsResult && prepareTransactionsResult.type !== 'possible') {
+      AppAnalytics.track(AppEvents.gas_fee_warning_impression, {
+        flow,
+        errorType: prepareTransactionsResult.type,
+        tokenId: feeCurrency.tokenId,
+        networkId: feeCurrency.networkId,
+      })
+    }
+  }, [flow, prepareTransactionsResult, feeCurrency])
+
+  const { title, description, ctaLabel, onPressCta } = useMemo(() => {
+    if (
+      !feeCurrency ||
+      !prepareTransactionsResult ||
+      prepareTransactionsResult.type === 'possible'
+    ) {
+      return {}
+    }
+    const title = t('gasFeeWarning.title', {
+      context: flow === 'Dapp' ? 'Dapp' : undefined,
```

Comment:
```suggestion
      context: flow
```
I think this should work the same, if a string for a value isn't found, it should fallback to the default one

Reply:
just tested and it does so updated it and updated unit tests